src/main/java/com/faforever/client/game/GameRunner.java

```diff
@@ -391,12 +391,10 @@ private Process launchOnlineGame(GameParameters gameParameters, Integer gpgPort,
   }
 
   private CompletableFuture<Integer> startIceAdapter(int uid) {
-    return iceAdapter.start(uid)
-                     .thenCompose(icePort -> coturnService.getSelectedCoturns(uid)
-                                                          .collectList()
-                                                          .doOnNext(iceAdapter::setIceServers)
-                                                          .thenReturn(icePort)
-                                                          .toFuture());
+    return coturnService.getIceSession(uid).toFuture().thenCompose(session -> {
+      iceAdapter.setIceServers(session.servers());
+      return iceAdapter.start(uid, session.forceRelay());
```

Comment:
I don't think the iceadapter will accept the ice servers being set before it starts since that is an rpc call

Reply:
Yeah wasn't aware its an rpc call, thought it just sets a property. Please check again.